### PR TITLE
P20 736 instrument lti metrics 2

### PR DESCRIPTION
**Pull request by nicklathe**
This PR instruments metrics from LTI events, based off [this doc](https://docs.google.com/document/d/19Yxl73BAA7lsA_s4s27rD2ITuvyb4U39iVSZETlDLOs/edit#heading=h.gdpxre740sd2). It includes metrics logged from both client and server side, of which both use the respective Statsig wrappers. Below are screen shots of what these events look like in the Statsig metrics dashboard.

**Client side event**
Screenshot 2024-03-25 at 1 50 03 PM

**Server side event**
Screenshot 2024-03-20 at 3 00 16 PM

TODO: I still need to set up a course with multiple sections in Canvas to verify the sync_course metrics are behaving as expected, however unit tests are passing.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/controllers/registrations_controller.rb
Comment (nicklathe) [inline dashboard/app/controllers/registrations_controller.rb]: Already did this, remove comment

File: dashboard/app/controllers/api/v1/sections_controller.rb
Comment (nicklathe) [inline dashboard/app/controllers/api/v1/sections_controller.rb:76]: NIT: consider changing this to `section.login_type`?

File: apps/src/lib/ui/lti/sync/LtiSectionSyncDialog.tsx
Comment (stephenliang) [inline apps/src/lib/ui/lti/sync/LtiSectionSyncDialog.tsx]: Should we move this to the success callback on line 101? I believe right now, this will mark `lti_opt_out_confirm` regardless if the API succeeded or not.

Comment (nicklathe) [reply apps/src/lib/ui/lti/sync/LtiSectionSyncDialog.tsx]: Ah yep, good call.

File: dashboard/app/controllers/lti_v1_controller.rb
Comment (nicklathe) [inline dashboard/app/controllers/lti_v1_controller.rb]: NIT: I might prefer this to be a one-liner like `total_students, total_sections = 0`.

File: apps/src/lib/ui/accounts/LtiRosterSyncSettings.jsx
Comment (carl-codeorg) [inline apps/src/lib/ui/accounts/LtiRosterSyncSettings.jsx]: Nit: this is really the lms_name not type

**Review by stephenliang — APPROVED** [review]

File: apps/src/lib/ui/lti/sync/LtiSectionSyncDialog.tsx
Comment (carl-codeorg) [inline apps/src/lib/ui/lti/sync/LtiSectionSyncDialog.tsx]: Do we have a preference for naming these events with a verb, like how the other opt out events are `toggle_on`, `toggle_off`, and `confirm`? If so this one should probably be `lti_opt_out_dialogue_view` or  `lti_opt_out_dialogue_show`

File: dashboard/app/controllers/lti_v1_controller.rb
Comment (carl-codeorg) [inline dashboard/app/controllers/lti_v1_controller.rb:307]: [This PR](https://github.com/code-dot-org/code-dot-org/pull/57563) fixes the way we count students during syncs. Right now, this will just count everything in the NRPS response, instead of the specific sections being synced (sections where the user isn't a teacher won't be synced). Your call if you want to add this now and have some numbers be slightly off, or hold off until after my PR. I could even just add it to mine.

**Review by carl-codeorg — COMMENTED** [review]
Looking good, just had a couple comments

File: apps/src/lib/ui/lti/sync/LtiSectionSyncDialog.tsx
Comment (nicklathe) [reply apps/src/lib/ui/lti/sync/LtiSectionSyncDialog.tsx]: I was working off Ali's [doc](https://docs.google.com/document/d/19Yxl73BAA7lsA_s4s27rD2ITuvyb4U39iVSZETlDLOs/edit#heading=h.gdpxre740sd2), however I want to circle back with him on some of these naming conventions. Even camel_casing these is a question, as I've seen other logs have just normal words with spacing.

File: apps/src/lib/ui/accounts/LtiRosterSyncSettings.jsx
Comment (nicklathe) [reply apps/src/lib/ui/accounts/LtiRosterSyncSettings.jsx]: You're right, I'm was going off Ali's doc. Will connect with him regarding naming of these values.

File: dashboard/app/controllers/lti_v1_controller.rb
Comment (nicklathe) [reply dashboard/app/controllers/lti_v1_controller.rb:307]: Sounds good, I'll rebase off your PR/staging once it gets merged and refactor this code. Looks like I'll just be able to get the section_count and student_count out of the returned sync result hash

Comment (carl-codeorg) [reply dashboard/app/controllers/lti_v1_controller.rb:307]: Just merged it in, so you should be good to rebase off staging

Comment (nicklathe) [reply dashboard/app/controllers/lti_v1_controller.rb:307]: Pulled in the changes and refactored this code.

File: apps/src/lib/ui/lti/sync/LtiSectionSyncDialog.tsx
Comment (nicklathe) [reply apps/src/lib/ui/lti/sync/LtiSectionSyncDialog.tsx]: I chatted with Ali a little this morning. Since he's been out, I haven't had a chance to sync with him on naming these events. I don't want to block this PR any longer, so I'd like to merge with the names as is, and make any potential changes in a follow-up PR.

**Review by carl-codeorg — APPROVED** [review]
:shipit:

**Pull request merged**
